src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -113,9 +114,9 @@ public SequencePosition Position
             get
             {
                 // TODO: Cannot use Slice even though it would be faster: https://github.com/dotnet/corefx/issues/33291
-                return _currentPosition.GetObject() == null
-                    ? default
-                    : _sequence.GetPosition(BytesConsumed);
+                return _isInputSequence
```

Comment:
There is no good way to distinguish between a default `ReadOnlySequence` and an empty `ReadOnlySequence`, so we are forced to use a bool to check what "mode" we are in. Returning a `default` SequencePosition would result in null ref exceptions if the user tries to use it in `ReadOnlySequence` APIs like `Slice`.

Reply:
> Returning a default SequencePosition would result in null ref exceptions if the user tries to use it in ReadOnlySequence APIs like Slice.

For a public struct that's meant to be used, that sounds like a bug.

Reply:
> For a public struct that's meant to be used, that sounds like a bug.

As far as I recall, we always had that behavior. If this isn't the appropriate behavior, we would need to fix it:
```C#
sequence = sequence.Slice(default(SequencePosition));
```
```text
System.NullReferenceException : Object reference not set to an instance of an object.
        Stack Trace:
          E:\GitHub\Fork\corefx\src\System.Memory\src\System\Buffers\ReadOnlySequence.Helpers.cs(243,0): at System.Buffers.ReadOnlySequence`1.BoundsCheck(SequencePosition& position)
          E:\GitHub\Fork\corefx\src\System.Memory\src\System\Buffers\ReadOnlySequence.cs(416,0): at System.Buffers.ReadOnlySequence`1.Slice(SequencePosition start)
```

For example, this is casting null:
https://github.com/dotnet/corefx/blob/45f4518def7d416eb7bd481d118209de2e703429/src/System.Memory/src/System/Buffers/ReadOnlySequence.Helpers.cs#L227-L249

cc @pakrym

Reply:
I mean, it really shouldn't null ref though...

Reply:
Yeah, this needs to be fixed. `default(ROS)` should behave like empty `ROS`.

Reply:
The concern is around `default(SequencePosition)`

Reply:
https://github.com/dotnet/corefx/issues/35254